### PR TITLE
fix: PacketBuffer.Clear() not locking packets

### PR DESCRIPTION
**Pull request by rpadrela**

### CODE REVIEW

File: src/lib/Connector.cs [hunk 1]
```diff
@@ -259,10 +258,7 @@ void WritePing(IOutOctetStream outStream)
         void SendChallenge(IOutOctetStream outStream)
         {
             var challenge = new ChallengeRequest(challengeNonce);
-            if (useDebugLogging)
-            {
-                log.Debug($"Sending Challenge {challenge}");
-            }
+            log.Debug($"Sending Challenge {challenge}");
```
Comment (eriksvedang) [inline src/lib/Connector.cs:261]: Do we really want to log this always? (I think all logging should be possible to turn off – though I'm not sure how the previous useDebugLogging actually was working)

File: src/lib/Connector.cs [hunk 2]
```diff
@@ -62,6 +62,7 @@ public class Connector : IPacketReceiver
         const byte NormalMode = 0x01;
         const byte OobMode = 0x02;
         const int MaxChallengeAttempts = 3;
+        const int MinExpectedPacketLength = 4;
```
Comment (eriksvedang) [inline src/lib/Connector.cs:65]: What's the story behind this specific number? :)

File: src/lib/Connector.cs [hunk 3]
```diff
@@ -283,36 +276,32 @@ void SendConnectRequest(IOutOctetStream outStream)
...
-            if (useDebugLogging)
-            {
-                log.Debug($"State: {newState} period:{period}");
-            }
+            log.Debug($"State: {newState} period:{period}");
 
             state = newState;
             stateChangeWait = period;
             lastStateChange = monotonicClock.NowMilliseconds();
+#if false
```
Comment (eriksvedang) [inline src/lib/Connector.cs:299]: What's up with this?

File: src/lib/Connector.cs [hunk 4]
```diff
@@ -432,13 +412,12 @@ void CheckDisconnect()
...
             {
-                if (useDebugLogging)
-                {
-                    log.Warning($"Disconnect detected! #{timeSinceReceivedHeader}");
-                }
+                log.Warning($"Disconnect detected! #{timeSinceReceivedHeader}");
 
                 SwitchState(ConnectionState.Disconnected, 9999);
-                //receiveStream.Lost();
+#if false
```
Comment (eriksvedang) [inline src/lib/Connector.cs:418]: ...and this?

File: src/lib/PacketRepository.cs [hunk 5]
```diff
@@ -37,7 +37,10 @@ public PacketBuffer(IMonotonicClock monotonicClock)
 
         public void Clear()
         {
-            packets.Clear();
+            lock (packets)
```
Comment (eriksvedang) [inline src/lib/PacketRepository.cs:40]: 👍

**Review by eriksvedang — COMMENTED** [review]
A bit confused by some of the changes. 
The `lock` seems like a great addition though!

File: src/lib/Connector.cs (hunk 1 above)
Comment (rpadrela) [reply src/lib/Connector.cs:261]: There should be no logic around the calls to the logger. We should log with the correct level (debug, trace, etc.) and then set the desired log level on the logger. That message will only be displayed on the log if we are logging with debug level or lower (e.g. trace, see https://github.com/piot/log-dotnet/blob/master/src/lib/LogLevel.cs).

File: src/lib/Connector.cs (hunk 2 above)
Comment (rpadrela) [reply src/lib/Connector.cs:65]: I don't know. It was used as a magic number but I have now given it a name.

File: src/lib/Connector.cs (hunk 1 above)
Comment (eriksvedang) [reply src/lib/Connector.cs:261]: Ah, I see. I misread and thought this was a call do `Debug.log`, but it was `log.Debug` :)

File: src/lib/Connector.cs (hunk 3 above)
Comment (rpadrela) [reply src/lib/Connector.cs:299]: Thought commented out code wasn't allowed and either removing it or ifdefing it out (if it's really helpful to be there) would be a better approach.

File: src/lib/Connector.cs (hunk 4 above)
Comment (rpadrela) [reply src/lib/Connector.cs:418]: Same as above.

File: src/lib/Connector.cs (hunk 3 above)
Comment (eriksvedang) [reply src/lib/Connector.cs:299]: OK, yeah I don't know why it's commented out...

File: src/lib/Connector.cs (hunk 2 above)
Comment (eriksvedang) [reply src/lib/Connector.cs:65]: Aha, I see :)

**Review by eriksvedang — APPROVED** [review]

**Pull request merged**